src/cargo/ops/cargo_test.rs

```diff
@@ -21,6 +22,30 @@ pub fn run_tests(
     let compilation = compile_tests(ws, options)?;
 
     if options.no_run {
+        let config = ws.config();
+        let cwd = config.cwd();
+        for UnitOutput {
+            unit,
+            path,
+            script_meta,
+        } in compilation.tests.iter()
+        {
+            let (exe_display, cmd) = cmd_builds(
+                config,
+                cwd,
+                unit,
+                path,
+                script_meta,
+                test_args,
+                &compilation,
+            )?;
+            config
+                .shell()
+                .concise(|shell| shell.status("Executable", &exe_display))?;
+            config
+                .shell()
+                .verbose(|shell| shell.status("Executable", &cmd))?;
+        }
```

Comment:
Since this is quite a lot of duplicated code with `run_benches`, can this be extracted out to a function to deduplicate it?